lib/nodejs/lib/thrift/http_connection.js

```diff
@@ -214,6 +214,18 @@ HttpConnection.prototype.write = function(data) {
   var req = (self.https) ?
       https.request(self.nodeOptions, self.responseCallback) :
       http.request(self.nodeOptions, self.responseCallback);
+
+  //support for timeout
+  var timeout = self.nodeOptions.timeout;
+  if(timeout){
+    req.on('socket', function (socket) {
+        socket.setTimeout(timeout);  
+        socket.on('timeout', function() {
+            req.abort();
+        });
+    });
+  }
```

Comment:
I cannot see the content of the patch as it has been removed.
It's visible to everyone while it's available.

I slightly prefer the different-name idea.
As to the actual name, I would choose "requestTimeout" especially if we're going to invoke `request.setTimeout`.
Seeing the [code](https://github.com/nodejs/node/blob/master/lib/_http_client.js#L501), the timeout callback is only active during each request life-cycle, and removed after completion so that it does not affect HTTP keep-alive behavior.
In other words, direct `socket.setTimeout` would break socket reuse.